include/spdk/blob.h

```diff
@@ -485,6 +485,15 @@ uint64_t spdk_blob_get_next_unallocated_io_unit(struct spdk_blob *blob, uint64_t
  */
 uint64_t spdk_blob_calc_used_clusters(struct spdk_blob *blob);
 
+/**
+ * Reset num_used_clusters_cache, if blob is thin provisioned and snapshot is
+ * created from it. New snapshot will own the data, hence the clusted information
+ * present in the blob cache can be cleared.
+ * \param blob Blob struct to query
+ * \return void
+ */
+void spdk_blob_reset_used_clusters_cache(struct spdk_blob *blob);
```

Comment:
My major concern is that we put extra step of resetting number of clusters on API users: they must explicitly perform such a call explicitly. I'd rather perform such step implicitly inside the SPDK.

Reply:
IMHO We can put it explicitly in the SPDK code, but possibly can't be put at the end of snapshot creation. So it is possible in some scenario We clearup cache but final snapshot creation can fail. If recalculation of alloc is not very expensive, then We can keep it in SPDK.